src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -295,43 +309,46 @@ private void BuildFinished(object sender, BuildFinishedEventArgs e)
 
         Terminal.BeginUpdate();
         try
-        {
-            string duration = (e.Timestamp - _buildStartTime).TotalSeconds.ToString("F1");
-            string buildResult = RenderBuildResult(e.Succeeded, _buildHasErrors, _buildHasWarnings);
-
-            Terminal.WriteLine("");
-            if (_restoreFailed)
+        { 
+            if (Verbosity > LoggerVerbosity.Quiet)
```

Comment:
Do we need the `BeginUpdate` and the finally block for the quiet mode? Or can we extract clearing of the stats into a  method - and for quiet just call that and return? (less nested method would get bit more readable)

Reply:
We need the finally block to execute there, and it needs calling of `BeginUpdate`. We might do as you suggest: put the finally block and clearing in a separate method, but I thought it would be easier to add the extra if condition to the messages I want to skip.